dessia_common/workflow/core.py

```diff
@@ -1941,6 +1943,78 @@ def _method_jsonschemas(self):
         warnings.warn("method_jsonschema method is deprecated. Use method_schema instead", DeprecationWarning)
         return self.method_schemas
 
+    def _split_input_values(self):
+        """ Split the input values for blocks and non-block variables. """
+        nbv_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if
+                      v in self.workflow.nonblock_variables]
+        block_values = [self.input_values[i] for i, v in enumerate(self.workflow.inputs) if
+                        v not in self.workflow.nonblock_variables]
+        return nbv_values, block_values
+
+    def _to_script(self, workflow_script):
+        """ Computes elements for a to_script interpretation. """
+        input_str = ""
+        default_value = ""
+        add_import = []
+
+        nbv_values, block_values = self._split_input_values()
+
+        # Blocks
+        index_ = 0
+        for j, block in enumerate(self.workflow.blocks):
+            for i, input_ in enumerate(block.inputs):
+                input_key = f"block_{j}.inputs[{i}]"
+                if input_key not in workflow_script.declaration:
+                    input_str += f"    workflow.input_index({input_key}): value_{j}_{i},\n"
+                    default_value += self._generate_default_value(block_values[index_], j, i)
+                    index_ += 1
+                    add_import.extend(update_imports(input_.type_, input_.name))
+
+        # NBVs
+        for k, nbv in enumerate(self.workflow.nonblock_variables):
+            input_str += f"    workflow.input_index(variable_{k}): value_{k},\n"
+            default_value += self._generate_default_value(nbv_values[k], k)
+            add_import.extend(update_imports(nbv.type_, nbv.name))
+
+        return input_str, default_value, add_import
+
+    def to_script(self):
+        """ Computes a script representing the WorkflowRun. """
+        workflow_script = self.workflow._to_script()
+        input_str, default_value, add_import = self._to_script(workflow_script)
+
+        workflow_script.declaration = f"{workflow_script.declaration}\n{default_value}\n\ninput_values =" \
+                                      f" {{\n{input_str}}}\nworkflow_run = workflow.run(input_values=input_values)\n"
+        workflow_script.imports.extend(add_import)
+
+        return workflow_script
+
+    @staticmethod
+    def _generate_default_value(value, block_index: int, input_index: int = None):
```

Comment:
Could this be a function, as well ? Instead of a static method

In addition to that, consider 
```py
input_index_str = "" if input_index is None else f"_{input_index}"
```

Reply:
yes,
done !